### PR TITLE
Use darker colors for app icons in app management.

### LINKED ISSUE #649: Use dark icons for app management
As a follow-up of https://github.com/nextcloud/server/pull/627 (white app menu with dark icons) it would be good for consistency if the app management would also use dark icons. Then we can also get rid of the strange grey box we needed to make the white icons visible.

@juliushaertl are you also up for this? ;) cc @nextcloud/designers

### PR DESCRIPTION
**Pull request by raghunayyar**
Should ideally fix https://github.com/nextcloud/server/issues/649, but the icons are coming as white for me, though I have changed the fills of all the icons. Can someone cross check if it is working fine for them? 

@jancborchardt I am not using #000 but #404040 which is the exact colour when you hover the icons in the app navigation. #000 cause a lot of contrast. :-)

Btw this is my first PR to NC ;-)

### CODE REVIEW

**Comment by juliushaertl** [comment]
@raghunayyar Thank you very much for your contribution. 

I think simply replacing the colors inside the SVGs will not work, as this would mean we need to do so in every app (also 3rd party ones). That means we'll lose compatibility to existing OwnCloud apps. See https://github.com/nextcloud/server/issues/649#issuecomment-238969952

@nextcloud/designers @jancborchardt what do you think?

**Comment by Bugsbane** [comment]
Personally, I would prefer to take a hybrid approach. For example, elsewhere it was previously suggested that we add an entry to the app info, specifically to point to a nextcloud specific app icon. OwnCloud simply wouldn't recognize this and would skip it. So we could look for a (dark) Nextcloud icon being set, and then if we didn't find it, fall back to taking the ownCloud one and run an SVG matrix filter over it like:

```
<feColorMatrix in="SourceGraphic" type="matrix" values="-1 0 0 0 1 
                                                        0 -1 0 0 1 
                                                        0 0 -1 0 1
                                                        0 0 0 1 0"/>
```

...to invert it. The only thing I'm not sure about is if there is sufficient cross browser support.

**Comment by raghunayyar** [comment]
I have updated the PR and re-implemented it with svg filters, though I am not liking it visually. Black on white (100% inversion) is creating way too much contrast. I would like @jancborchardt / @nextcloud/designers  to take the call. Here is a screenshot:
screen shot 2016-08-11 at 1 01 32 pm

**Comment by MorrisJobke** [comment]
> Black on white (100% inversion) is creating way too much contrast.

Then simply add a opacity of 60% to it via CSS 🙈 😉

**Comment by MorrisJobke** [comment]
bildschirmfoto 2016-08-11 um 10 46 10

**Comment by raghunayyar** [comment]
@MorrisJobke done. All yours. ;)

**Comment by juliushaertl** [comment]
@raghunayyar Looks good so far, just one more thing: For app list coming from the app store, there is a preview of the apps shown. We need to make sure, that this previews don't get inverted like the icons.

2016-08-11-125607_354x173_scrot

**Comment by jancborchardt** [comment]
Great progress here! Looks good except for the issue @juliushaertl already noted. :)

**Comment by raghunayyar** [comment]
Will fix this one today.

**Comment by raghunayyar** [comment]
Fixed, can we review this. :)

**Comment by juliushaertl** [comment]
looks good now :+1: 

cc @nextcloud/designers for another reviewer

**Comment by mar1u5** [comment]
:+1:

**Comment by mar1u5** [comment]
We want to have this in `stable10` right? @karlitschek

**Comment by karlitschek** [comment]
nice. please backport

**Comment by Bugsbane** [comment]
BTW Congrats on your first PR @raghunayyar !

**Comment by MorrisJobke** [comment]
I will create the backport PR :)

> BTW Congrats on your first PR @raghunayyar !

Yes! 🎉

**Comment by mar1u5** [comment]
Haha... Forgotten to backport 😁 Sorry @MorrisJobke 😅

File: settings/js/apps.js [hunk 1]
```diff
@@ -226,6 +226,23 @@ OC.Settings.Apps = OC.Settings.Apps || {
...
+	 * appfromstore: bool to check whether the app is fetched from store or not.
+	 */
+
+	imageUrl : function (url, appfromstore) {
+		var img = '<svg width="72" height="72" viewBox="0 0 72 72">';
+		if (appfromstore) {
+			img += '<image x="0" y="0" width="72" height="72" preserveAspectRatio="xMinYMin meet" xlink:href="' + url + '"  class="app-icon" /></svg>';
+		} else {
+			img += '<defs><filter id="invert"><feColorMatrix in="SourceGraphic" type="matrix" values="-1 0 0 0 1 0 -1 0 0 1 0 0 -1 0 1 0 0 0 1 0" /></filter></defs>';
+			img += '<image x="0" y="0" width="72" height="72" preserveAspectRatio="xMinYMin meet" filter="url(#invert)" xlink:href="' + url + '"  class="app-icon" /></svg>';
```
Comment (MorrisJobke) [inline settings/js/apps.js]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/image *cough* *cough* ... this breaks the app managment in all browsers except chrome (where it only works by luck). I will revert it and we better load the SVG directly into the DOM which is even supported in IE9 and makes CSS styling easier too.

Comment (MorrisJobke) [inline settings/js/apps.js]: Ah just noticed that this is wrapped inside an SVG ... then I will try to fix this.

Comment (MorrisJobke) [inline settings/js/apps.js]: https://developer.mozilla.org/en-US/docs/Web/SVG/Element/image

**Comment by MorrisJobke** [comment]
Regression: #900

**Pull request merged**